### ISSUE TITLE
Better error messages in common cases

### ISSUE THREAD

**Issue by tibbetts** [probcomp/bayeslite#165]
In the notebook environment, running queries with our q() helper, we get big backtraces and hard to follow errors for simple things.

One solution might be to have our own iPython kernel.

Another would be to have our helper method (or some decorator it calls) turn certain exceptions into warning.error() messages rather than raw backtraces.

**Comment by marcoct** [comment]
The above seems to deal more with iPython notebook, but the message content of parse error exceptions may also benefit from more detail. For example, forgetting to provide the from satellites_cc:
`infer explicit name, purpose, predict purpose as pp confidence pp_conf`
results in this being raised: `BQLParseError: Parse error: syntax error near ;`

**Comment by riastradh-probcomp** [comment]
It would be nice to give better parse error messages.  There are a few obvious ways to go about doing this:

- Do a better job particularly with semicolon, which did not actually appear in the user's input stream, but was implicitly added at the end of string for expedience.
- Add grammar rules that guess what invalid syntax people might use, and write parser actions that report more specific syntax errors.
- Adapt lemonade to report the set of all tokens it expected to see in the place where a syntax error appeared, and list those tokens.

**Comment by gregory-marton** [comment]
https://github.com/probcomp/bayeslite/issues/174 seems to be a specific instance of this kind of bad error message. If we caught common exceptions and displayed them in a friendlier way, then the upstream bug that fails to properly process some backtraces would have a much-diminished effect.

**Comment by gregory-marton** [comment]
2448819
